--- a/extract/ExtractTypes.ml
+++ b/extract/ExtractTypes.ml
@@ -9,11 +9,10 @@
 
 let extract_str (_span : Meta.span) (fmt : F.formatter) ~(inside : bool)
     (s : string) : unit =
-  let chars = StringUtils.string_to_chars s in
   (* Using the OCaml escape conventions for now.
 
      TODO: does this really work? *)
-  let s = String.concat "" (List.map Char.escaped chars) in
+  let s = String.escaped s in
   let s = "\"" ^ s ^ "\"" in
 
   (* We need to convert the string to a str (the conversion inserts a static check
